fili-core/src/test/groovy/com/yahoo/bard/webservice/table/resolver/DimensionIdFilterSpec.groovy

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.resolver
+
+import com.yahoo.bard.webservice.data.dimension.Dimension
+import com.yahoo.bard.webservice.data.dimension.DimensionField
+import com.yahoo.bard.webservice.data.dimension.SearchProvider
+import com.yahoo.bard.webservice.web.ApiFilter
+import com.yahoo.bard.webservice.web.FilterOperation
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class DimensionIdFilterSpec extends Specification {
+
+    def "Constructor builds the expected ApiFilter map"() {
+        setup:
+        def (Dimension dimension, _, ApiFilter expectedApiFilter, DimensionIdFilter dimensionIdFilter) = buildSimpleTestData()
+
+        expect:
+        dimensionIdFilter.dimensionKeySelectFilters == [(dimension): expectedApiFilter]
+    }
+
+    def "anyRowsMatch() throws IllegalArgumentException if dimension is not defined on the DimensionIdFilter"() {
+        def (_, foo, bar, DimensionIdFilter dimensionIdFilter) = buildSimpleTestData()
+
+        when:
+        dimensionIdFilter.anyRowsMatch(Mock(Dimension), null)
+
+        then:
+        thrown(IllegalArgumentException)
+    }
+
+    def "anyRowsMatch() appends apiFilter from DimensionIdFilter to constraint filters and queries search provider"() {
+        given: "A simple dimension id filter"
+        def (Dimension dimension, SearchProvider searchProvider, ApiFilter dimensionIdApiFilter, DimensionIdFilter filter) = buildSimpleTestData()
+
+        DimensionField keyField = dimension.getKey()
+
+        and: "a search provider expecting concatenated api filters"
+        ApiFilter constraintApiFilter = new ApiFilter(dimension, keyField, FilterOperation.in, ["a"] as Set)
+        Set<ApiFilter> expectedSearchApiFilters = [dimensionIdApiFilter, constraintApiFilter] as HashSet
+
+        1 * searchProvider.hasAnyRows(expectedSearchApiFilters) >> true
+
+        expect: "Check that the expected values are submitted to the backing search provider"
+        filter.anyRowsMatch(dimension, [constraintApiFilter] as Set)
+    }
+
+
+    @Unroll
+    def "Does emptyConstraintOrAnyRows match return #expected with #description"() {
+        given: "A constraint map with some constraints"
+        def (Dimension dimension, SearchProvider searchProvider, Object foo, DimensionIdFilter filter) = buildSimpleTestData()
+
+        Map<Dimension, Set<ApiFilter>> constraintMap = hasApiFilter ?
+                [(dimension): (emptyApiFilter ? Collections.emptySet() : [Mock(ApiFilter)] as Set)] :
+                [:]
+
+        searchProvider.hasAnyRows(_) >> searchRowsExist
+
+        expect: "filter matches when there are no constraints or they exist and return rows from search provider"
+        expected == filter.emptyConstraintOrAnyRows(dimension, constraintMap)
```

Comment:
Expected should be on the right hand side. IDE will display result on the left hand side as expected I think.